http/response.go

```diff
@@ -67,13 +70,29 @@ func (res *Response) RawNext() (interface{}, error) {
 
 	m := &cmds.MaybeError{Value: res.req.Command.Type}
 	err := res.dec.Decode(m)
+	if err != nil {
+		if res.err != nil && err.Error() == res.err.Error() {
+			// last error was sent as value, now we get the same error from the headers. ignore and EOF!
+			err = io.EOF
+		} else if err == io.EOF {
+			// just return EOF errors
+			return nil, err
+		} else {
+			// wrap all other errors
+			res.err = &cmdkit.Error{Message: err.Error()}
+		}
+	}
 
-	// last error was sent as value, now we get the same error from the headers. ignore and EOF!
-	if err != nil && res.err != nil && err.Error() == res.err.Error() {
-		err = io.EOF
+	v, err := m.Get()
+	if err != nil {
+		if e, ok := err.(*cmdkit.Error); ok {
+			res.err = e
+		} else {
+			res.err = &cmdkit.Error{Message: err.Error()}
+		}
 	}
 
-	return m.Get(), err
+	return v, err
 }
 
 func (res *Response) Next() (interface{}, error) {
```

Comment:
THis function checks if the value is an error, I assume this isn't possible given RawNext?